### ISSUE TITLE
cm 10.2: Wifi is disturbing other wifi clients

### ISSUE THREAD

**Issue by the-dreamer** [LegacyXperia/local_manifests#20]
When i transfer a file from my notebook i can see the thruput (win 8) and see a very unstable value when wifi is enabled on my xperia ray. Values range is from 1,5 MB/s to 0 MB/s. The Ray is in deep standby and has no active data transmission. 
When wifi is disabled on ray the thruput is stable round about 2,2 MB/s to 1,7 MB/s

When the Notebook is in a room with less signal strenght the connection is completely disturbed and no data transfer is possible at all. But the connection itself to the wifi network is still there. 

This behaviour is intruduced with the cm 10.2 first alpha image and still there on the second alpha :-/

**Comment by olivierboro** [comment]
Same for me (xperia arc lt15i) : when I activate Wifi on the phone, the Wifi speed on my macbook pro drops from 120Mb to 5Mb and become unusable.

**Comment by jlmcr87** [comment]
Same problem for me

**Comment by soulpower11** [comment]
Happens to me too (Xperia Neo V) Have the same issue when using cm10.1 too so it might be a kernel problem

**Comment by b-bremer** [comment]
Same on pro and cm-10.2_iyokan-LX-20130819-NIGHTLY

**Comment by GianniDPC** [comment]
I'm having the same issues..., Using the lasted CM 10.2 Alpha, And CM 10.1 too.

**Comment by enli** [comment]
I confirm this. I am on CM10.1 Ray. Whenever Ray is connected to Wi-fi, my two laptops connectivity suffer a lot. Usually I get 0% packet loss on those two laptops but when Ray joins the network, I get 10-20% packet loss on both of the laptops. I guess this has started happening since I switched to nAa kernels.

**Comment by pcfighter** [comment]
So it's another reason to re-write WiFi implementation on our board (first reason is WiFi interruptions). Our team working on it. I hope this issue will be resolved with fixing wifi interruptions.

**Comment by mikeNG** [comment]
Wait for the next release and report back

**Comment by hawrehas** [comment]
thanks allot mike and pcfighter! seems like the new build (10.2 build 20130918) fixed the problem for smultron.
can anyone confirm? (It's hard to tell for sure if it's fixed because I don't know exactly how to reproduce it)

Edit: tried a speedtest on laptop while browsing with my mobile in wifi. No big impact on speed :) with the previous version even google didn't work properly on my laptop like you described. So for me seems to be fixed (for now... maybe there are other variables)

**Comment by jlmcr87** [comment]
Dear Mike, pcfighter, etc....the problem persist with the latest build 20130918. Very slow connection on all my pcs, when i connect the phone to my wifi. So slow that it takes a lot to open google.

**Comment by jlmcr87** [comment]
Full Log (kernel, modem, and main log) it contains power on the phone and connect and disconnect to my wifi. Hope it helps, download it here (25kb only):
http://www.mediafire.com/?lv7ll8fq8zx88a4

**Comment by the-dreamer** [comment]
Confirmed: Wifi still seems to disturbing other clients. Currently i cannot test it with win 8 because my server is crashed yesterday. But Websites have problems loading when wifi is enabled on my ray. Just for information: i have a pure b/g wireless network running.

**Comment by soulpower11** [comment]
It is still happening to me too.

**Comment by Klami** [comment]
i have the same problems... i've tested a lot and found a solution for me... I set the W-LAN Channel in the Router under 5 and everything work's fine by me.

Device:
anzu
cm10.2(19/08/2013)

**Comment by siankatabg** [comment]
I think that i found the answer.Can somebody try to put this http://www.mediafire.com/?5go6gur2d3x3s9g in /system/etc/firmware/ti-connectivity with right permissions,restart phone and tell us if its good?

**Comment by the-dreamer** [comment]
What have you changed? Or what could be the reason?

**Comment by jlmcr87** [comment]
siankatabg your archive fix the bug. Please contact Mike to include it in the next release. Big Thanks!!!!!!!!!!!

**Comment by mikeNG** [comment]
@siankatabg which version is this firmware? and where did you find it?

**Comment by jlmcr87** [comment]
I don´t know what siankatabg have done, but it just fix the bug. Now I can use my wifi!!!!!!!!!

**Comment by mikeNG** [comment]
can you post a dmesg when you enable wifi? so I can see which firmware version it loads?

**Comment by mikeNG** [comment]
Current firmware:

```
<6>[  812.688232] wl12xx: firmware booted (Rev 6.3.10.0.125)
```

New firmware:

```
<6>[  937.688812] wl12xx: firmware booted (Rev 6.3.10.0.136)
```

**Comment by jlmcr87** [comment]
So it was a problem of the wifi firmware? Can you explain what happen? After a couple of months having this issue it will be good you explain it a little! Thanks!

**Comment by mikeNG** [comment]
I don't know what happened because the firmware is not open source. By the way, this firmware is not meant for the current wifi source we are using, it's meant to be used with newer source from TI. I will test this for a few hours and probably merge it.

**Comment by siankatabg** [comment]
All was experiment :D But I'm glad I could help.

Sorry for my English.

**Comment by jlmcr87** [comment]
Testing it and now wifi is perfect (battery drain is 6-7% with 95% deep sleep). Also i think data interruptions (include 2G data) have disappeared. Is this possible? I need to test that more

**Comment by mikeNG** [comment]
there is even newer firmware available, 6.3.10.0.137. I will test this

**Comment by mikeNG** [comment]
Fixed in next build.
